packages/notebook/src/widget.ts

```diff
@@ -892,6 +900,44 @@ export class StaticNotebook extends WindowedList {
     }
   }
 
+  private async _updateForDeferMode(
+    cell: Cell<ICellModel>,
+    cellIdx: number
+  ): Promise<void> {
+    cell.dataset.windowedListIndex = `${cellIdx}`;
+    this.layout.insertWidget(cellIdx, cell);
+    await cell.ready;
+  }
+
+  private _renderCSSAndJSOutputs(
+    cell: Cell<ICellModel>,
+    cellIdx: number
+  ): void {
+    // Only render cell with text/html outputs containing scripts or/and styles
+    // Note:
+    //   We don't need to render JavaScript mimetype outputs because they get
+    //   directly evaluate without adding DOM elements (see @jupyterlab/javascript-extension)
+    if (cell instanceof CodeCell) {
+      for (
+        let outputIdx = 0;
+        outputIdx < cell.model.outputs.length;
+        outputIdx++
+      ) {
+        const output = cell.model.outputs.get(outputIdx);
+        const html =
```

Comment:
Should we also check for `javaScriptRendererFactory` and render it unconditionally? For example, in IPython execute:

```
%%javascript
alert('test')
```

Reply:
We don't need it per the above comment (line 917); - to explicit it here - it already works as the JavaScript will not be injected as a `script` element in the DOM but directly evaluated.

> This is tested in the updated `windowed_notebook.ipynb`.